### PR TITLE
Int b 20975 too submits mto and receives incorrect validation

### PR DESCRIPTION
**Pull request by samaysofo**
## Summary
When the MTO shipment is submitted by the TOO, the confirmation message on the screen reads "Customer created successfully". The TOO does not create the customer, so this changes are made to display "Customer created successfully" after a SC saves a customer profile and also display "Task order created successfully" when a TOO approves a move. 
Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [x] Has the branch been pulled in and checked out?
- [x] Have the BL acceptance criteria been met for this change?
- [x] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the MillMove application
2. Login in with an account with multiple roles at least with (Service Counselor(SC)  and Task Ordering Office(TOO) )
3. SC creates a new customer account , receives "Customer created successfully". Then go and create a move.
4. Log in as a TOO
5. Review and Submit move
6. "Task order created successfully" confirmation message displays.
### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
image
image

### CODE REVIEW

**Comment by brianmanley-caci** [comment]
One of the integration test is failing, but I'm not sure if it is related to your changes.

**Comment by samaysofo** [comment]
> One of the integration test is failing, but I'm not sure if it is related to your changes.
I took a look but doesn't look like its related to the changes i made but i could be wrong, @danieljordan-caci  @r-mettler do you by chance know if this is one of those flaky failed tests?

**Comment by danieljordan-caci** [comment]
> > One of the integration test is failing, but I'm not sure if it is related to your changes.
> > I took a look but doesn't look like its related to the changes i made but i could be wrong, @danieljordan-caci  @r-mettler do you by chance know if this is one of those flaky failed tests?

It is a flaky test so you can disregard

**Review by brianmanley-caci — APPROVED** [review]

**Comment by samaysofo** [comment]
@danieljordan-caci / @r-mettler may you please take a look at this when you get the chance? thanks

**Comment by danieljordan-caci** [comment]
> @danieljordan-caci / @r-mettler may you please take a look at this when you get the chance? thanks

Looking now

**Review by ryan-mchugh — APPROVED** [review]

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
So... I am seeing the `Task order created successfully` message, but when creating the customer I am being redirected. I know WHY this is (see [this PR](https://github.com/transcom/mymove/pull/13797)) and we likely need to allow the SC access to that add orders route when coming from the create customer process. Similar to how we allow the SC to access the add orders route when coming from the `Create New Move` button.

Task order flash message
Screenshot 2024-10-08 at 1 55 22 PM

https://github.com/user-attachments/assets/c58621d8-ad0c-43c4-bd15-aa70c8409978

**Comment by samaysofo** [comment]
> So... I am seeing the `Task order created successfully` message, but when creating the customer I am being redirected. I know WHY this is (see [this PR](https://github.com/transcom/mymove/pull/13797)) and we likely need to allow the SC access to that add orders route when coming from the create customer process. Similar to how we allow the SC to access the add orders route when coming from the `Create New Move` button.
> 
> Task order flash message Screenshot 2024-10-08 at 1 55 22 PM
> 
>  Screen.Recording.2024-10-08.at.2.00.12.PM.mov

**Comment by samaysofo** [comment]
> So... I am seeing the `Task order created successfully` message, but when creating the customer I am being redirected. I know WHY this is (see [this PR](https://github.com/transcom/mymove/pull/13797)) and we likely need to allow the SC access to that add orders route when coming from the create customer process. Similar to how we allow the SC to access the add orders route when coming from the `Create New Move` button.
> 
> Task order flash message Screenshot 2024-10-08 at 1 55 22 PM
> 
>  Screen.Recording.2024-10-08.at.2.00.12.PM.mov

Ok, will take a look at it . it works fine when yes is clicked for "Okta Account" radio button. So i'm assuming that's a different call when "no" is clicked?

**Comment by danieljordan-caci** [comment]
> > So... I am seeing the `Task order created successfully` message, but when creating the customer I am being redirected. I know WHY this is (see [this PR](https://github.com/transcom/mymove/pull/13797)) and we likely need to allow the SC access to that add orders route when coming from the create customer process. Similar to how we allow the SC to access the add orders route when coming from the `Create New Move` button.
> > Task order flash message Screenshot 2024-10-08 at 1 55 22 PM
> > Screen.Recording.2024-10-08.at.2.00.12.PM.mov
> 
> Ok, will take a look at it . it works fine when yes is clicked for "Okta Account" radio button. So i'm assuming that's a different call when "no" is clicked?

It is not a different API call. Shouldn't matter anyway per the code:
```
    return createCustomerWithOktaOption({ body })
      .then((res) => {
        const customerId = Object.keys(res.createdCustomer)[0];
        setFlashMessage('CUSTOMER_CREATE_SUCCESS', 'success', `Customer created successfully.`);
        navigate(
          generatePath(servicesCounselingRoutes.BASE_CUSTOMERS_ORDERS_ADD_PATH, {
            customerId,
          }),
          { state: { isSafetyMoveSelected: isSafetyMove } },
        );
      })
      .catch((e) => {
        let errorMessage;
        if (e.status === 409) errorMessage = 'This EMPLID is already in use';
        else errorMessage = getResponseError(e?.response, 'failed to create service member due to server error');
        setServerError(errorMessage);
      });
```

**Comment by samaysofo** [comment]
> > > So... I am seeing the `Task order created successfully` message, but when creating the customer I am being redirected. I know WHY this is (see [this PR](https://github.com/transcom/mymove/pull/13797)) and we likely need to allow the SC access to that add orders route when coming from the create customer process. Similar to how we allow the SC to access the add orders route when coming from the `Create New Move` button.
> > > Task order flash message Screenshot 2024-10-08 at 1 55 22 PM
> > > Screen.Recording.2024-10-08.at.2.00.12.PM.mov
> > 
> > 
> > Ok, will take a look at it . it works fine when yes is clicked for "Okta Account" radio button. So i'm assuming that's a different call when "no" is clicked?
> 
> It is not a different API call. Shouldn't matter anyway per the code:
> 
> ```
>     return createCustomerWithOktaOption({ body })
>       .then((res) => {
>         const customerId = Object.keys(res.createdCustomer)[0];
>         setFlashMessage('CUSTOMER_CREATE_SUCCESS', 'success', `Customer created successfully.`);
>         navigate(
>           generatePath(servicesCounselingRoutes.BASE_CUSTOMERS_ORDERS_ADD_PATH, {
>             customerId,
>           }),
>           { state: { isSafetyMoveSelected: isSafetyMove } },
>         );
>       })
>       .catch((e) => {
>         let errorMessage;
>         if (e.status === 409) errorMessage = 'This EMPLID is already in use';
>         else errorMessage = getResponseError(e?.response, 'failed to create service member due to server error');
>         setServerError(errorMessage);
>       });
> ```

ok, so out of curiosity. I had to double check both the main/int branch and the main branch works fine but the int branch does the redirecting. 

https://github.com/user-attachments/assets/e3bcb8a3-f607-42ca-a2f0-dcc7ee1b32bf

https://github.com/user-attachments/assets/f33366cd-3451-4e41-b411-7165d78d2dde

**Comment by danieljordan-caci** [comment]
> > > > So... I am seeing the `Task order created successfully` message, but when creating the customer I am being redirected. I know WHY this is (see [this PR](https://github.com/transcom/mymove/pull/13797)) and we likely need to allow the SC access to that add orders route when coming from the create customer process. Similar to how we allow the SC to access the add orders route when coming from the `Create New Move` button.
> > > > Task order flash message Screenshot 2024-10-08 at 1 55 22 PM
> > > > Screen.Recording.2024-10-08.at.2.00.12.PM.mov
> > > 
> > > 
> > > Ok, will take a look at it . it works fine when yes is clicked for "Okta Account" radio button. So i'm assuming that's a different call when "no" is clicked?
> > 
> > 
> > It is not a different API call. Shouldn't matter anyway per the code:
> > ```
> >     return createCustomerWithOktaOption({ body })
> >       .then((res) => {
> >         const customerId = Object.keys(res.createdCustomer)[0];
> >         setFlashMessage('CUSTOMER_CREATE_SUCCESS', 'success', `Customer created successfully.`);
> >         navigate(
> >           generatePath(servicesCounselingRoutes.BASE_CUSTOMERS_ORDERS_ADD_PATH, {
> >             customerId,
> >           }),
> >           { state: { isSafetyMoveSelected: isSafetyMove } },
> >         );
> >       })
> >       .catch((e) => {
> >         let errorMessage;
> >         if (e.status === 409) errorMessage = 'This EMPLID is already in use';
> >         else errorMessage = getResponseError(e?.response, 'failed to create service member due to server error');
> >         setServerError(errorMessage);
> >       });
> > ```
> 
> ok, so out of curiosity. I had to double check both the main/int branch and the main branch works fine but the int branch does the redirecting.
> 
>  create_cust001_main.mov 
>  cust_creation_int.mov

If you pull the newest `main` you'll see that it should now do what `integrationTesting` does :)

[Just merged it](https://github.com/transcom/mymove/pull/13798)

**Comment by samaysofo** [comment]
> > > > > So... I am seeing the `Task order created successfully` message, but when creating the customer I am being redirected. I know WHY this is (see [this PR](https://github.com/transcom/mymove/pull/13797)) and we likely need to allow the SC access to that add orders route when coming from the create customer process. Similar to how we allow the SC to access the add orders route when coming from the `Create New Move` button.
> > > > > Task order flash message Screenshot 2024-10-08 at 1 55 22 PM
> > > > > Screen.Recording.2024-10-08.at.2.00.12.PM.mov
> > > > 
> > > > 
> > > > Ok, will take a look at it . it works fine when yes is clicked for "Okta Account" radio button. So i'm assuming that's a different call when "no" is clicked?
> > > 
> > > 
> > > It is not a different API call. Shouldn't matter anyway per the code:
> > > ```
> > >     return createCustomerWithOktaOption({ body })
> > >       .then((res) => {
> > >         const customerId = Object.keys(res.createdCustomer)[0];
> > >         setFlashMessage('CUSTOMER_CREATE_SUCCESS', 'success', `Customer created successfully.`);
> > >         navigate(
> > >           generatePath(servicesCounselingRoutes.BASE_CUSTOMERS_ORDERS_ADD_PATH, {
> > >             customerId,
> > >           }),
> > >           { state: { isSafetyMoveSelected: isSafetyMove } },
> > >         );
> > >       })
> > >       .catch((e) => {
> > >         let errorMessage;
> > >         if (e.status === 409) errorMessage = 'This EMPLID is already in use';
> > >         else errorMessage = getResponseError(e?.response, 'failed to create service member due to server error');
> > >         setServerError(errorMessage);
> > >       });
> > > ```
> > 
> > 
> > ok, so out of curiosity. I had to double check both the main/int branch and the main branch works fine but the int branch does the redirecting.
> > create_cust001_main.mov
> > cust_creation_int.mov
> 
> If you pull the newest `main` you'll see that it should now do what `integrationTesting` does :)
> 
> [Just merged it](https://github.com/transcom/mymove/pull/13798)

The pr has been updated with the fix. 

https://github.com/user-attachments/assets/425f70d3-d6ea-4e1e-971b-bb0bedfc6083

**Comment by msaki-caci** [comment]
@samaysofo I think something may have went wrong with your latest push. I am showing 236 files changed right now. Is that right?

image

**Comment by danieljordan-caci** [comment]
> @samaysofo I think something may have went wrong with your latest push. I am showing 236 files changed right now. Is that right?
> 
> image

Needed to merge in the latest `integrationTesting` - we were seeing the Go upgrade file changes, which were not in this branch. Should be fixed now - just updated the branch.

**Comment by samaysofo** [comment]
> @samaysofo I think something may have went wrong with your latest push. I am showing 236 files changed right now. Is that right?
> 
> image

i probably need to update my branch with the latest integration branch.

**Comment by msaki-caci** [comment]
> > @samaysofo I think something may have went wrong with your latest push. I am showing 236 files changed right now. Is that right?
> > image
> 
> i probably need to update my branch with the latest integration branch.

Yeah its all good now 👍

**Comment by msaki-caci** [comment]
LGTM but you have a failure in one of the circleci tests

**Comment by samaysofo** [comment]
> LGTM but you have a failure in one of the circleci tests
I still see some changes in the pr that don't belong here in pkg/services/mocks/ which seems to be what's failing. Let me update my branch with INT again, not sure why they are still there after the update.

**Comment by danieljordan-caci** [comment]
> > LGTM but you have a failure in one of the circleci tests
> > I still see some changes in the pr that don't belong here in pkg/services/mocks/ which seems to be what's failing. Let me update my branch with INT again, not sure why they are still there after the update.

On your branch run `make mocks_generate` and push the changes

**Comment by samaysofo** [comment]
> > > LGTM but you have a failure in one of the circleci tests
> > > I still see some changes in the pr that don't belong here in pkg/services/mocks/ which seems to be what's failing. Let me update my branch with INT again, not sure why they are still there after the update.
> 
> On your branch run `make mocks_generate` and push the changes

@danieljordan-caci @msaki-caci its cleanup up now, thanks.

**Review by msaki-caci — APPROVED** [review]

File: src/pages/Office/CustomerOnboarding/CreateCustomerForm.jsx [hunk 1]
```diff
@@ -133,6 +134,7 @@ export const CreateCustomerForm = ({ userPrivileges, setFlashMessage }) => {
     return createCustomerWithOktaOption({ body })
       .then((res) => {
         const customerId = Object.keys(res.createdCustomer)[0];
+        setCanAddOrders(true);
         setFlashMessage('CUSTOMER_CREATE_SUCCESS', 'success', `Customer created successfully.`);
         navigate(
           generatePath(servicesCounselingRoutes.BASE_CUSTOMERS_ORDERS_ADD_PATH, {
```
Comment (danieljordan-caci) [inline src/pages/Office/CustomerOnboarding/CreateCustomerForm.jsx:140]: Can you add/update a test to reflect these changes? Thanks.

Comment (samaysofo) [reply src/pages/Office/CustomerOnboarding/CreateCustomerForm.jsx:140]: test updated.

**Review by danieljordan-caci — APPROVED** [review]
Most excellent!

**Comment by samaysofo** [comment]
> Most excellent!

@danieljordan-caci i see some server test failing which seems to have nothing to do with my changes after i updated my branch again with INT before i merge, do you think i'm still good to merge?

**Pull request merged**